src/toil_scripts/lib/urls.py

```diff
@@ -0,0 +1,147 @@
+import base64
+import hashlib
+import os
+import subprocess
+from urlparse import urlparse
+
+
+def download_url(url, work_dir='.', name=None, s3_encryption_key_path=None):
+    """
+    Downloads URL, can pass in file://, http://, s3://, or ftp://
+
+    :param str url: URL to download from
+    :param str work_dir: Directory to download file to
+    :param str name: Name of output file, if None, basename of URL is used
+    :param str s3_encryption_key_path: Path to 32-byte encryption key if url points to S3 file that uses SSE-C
+    :return str: Path to the downloaded file
+    """
+    file_path = os.path.join(work_dir, name) if name else os.path.join(work_dir, os.path.basename(url))
+    if s3_encryption_key_path:
+        _download_encrypted_file(url, file_path, s3_encryption_key_path)
+    elif url.startswith('s3:'):
+        _download_s3_url(file_path, url)
+    else:
+        subprocess.check_call(['curl', '-fs', '--retry', '5', '--create-dir', url, '-o', file_path])
+    assert os.path.exists(file_path)
```

Comment:
Sorry for the confusion. I wasn't suggesting that it should be removed (for the exact reason you stated), but rather that we document why this might happen (if there is an explicit reason that we expect this assert might fire).